backend/config/db_router.py

```
@@ -1,101 +1,211 @@
+                conn = connections[alias]
+                with conn.cursor() as cursor:
+                    # PostgreSQL: pg_stat_replication or pg_last_xact_replay_timestamp
+                    cursor.execute(
+                        "SELECT EXTRACT(EPOCH FROM (now() - pg_last_xact_replay_timestamp()))::FLOAT"
+                    )
+                    row = cursor.fetchone()
+                    lag = round(row[0], 3) if row and row[0] is not None else 0.0
+                    info["lag_seconds"] = lag
+                    threshold = getattr(settings, "REPLICA_LAG_ALERT_SECONDS", 30)
+                    info["status"] = "lagging" if lag > threshold else "healthy"
+            except Exception as exc:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)